frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/EditTab/OptionListEditor/OptionListEditor.tsx

```diff
@@ -1,39 +1,42 @@
-import React, { createRef, useRef, useState } from 'react';
-import type { Option } from 'app-shared/types/Option';
+import React from 'react';
 import { useTranslation } from 'react-i18next';
-import {
-  StudioCodeListEditor,
-  StudioModal,
-  StudioSpinner,
-  StudioErrorMessage,
-  StudioAlert,
-  StudioProperty,
-} from '@studio/components';
-import type { CodeListEditorTexts } from '@studio/components';
+import { StudioSpinner, StudioErrorMessage, StudioAlert } from '@studio/components';
 import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
-import { useUpdateOptionListMutation } from 'app-shared/hooks/mutations/useUpdateOptionListMutation';
 import { useOptionListsQuery } from 'app-shared/hooks/queries/useOptionListsQuery';
-import { useOptionListEditorTexts } from '../../hooks/useOptionListEditorTexts';
-import { usePreviewContext } from 'app-development/contexts/PreviewContext';
 import type { IGenericEditComponent } from '../../../../../componentConfig';
 import type { SelectionComponentType } from '../../../../../../../types/FormComponent';
-import classes from './OptionListEditor.module.css';
+import { LibraryOptionsEditor } from './LibraryOptionsEditor';
+import { ManualOptionsEditor } from './ManualOptionsEditor';
 
-type OptionListEditorProps = {
+export type OptionListEditorProps = {
   optionsId: string;
-  label: string;
+  setComponentHasOptionList: (value: boolean) => void;
 } & Pick<IGenericEditComponent<SelectionComponentType>, 'component' | 'handleComponentChange'>;
 
 export function OptionListEditor({
   optionsId,
   component,
   handleComponentChange,
-  label,
+  setComponentHasOptionList,
 }: OptionListEditorProps): React.ReactNode {
   const { t } = useTranslation();
   const { org, app } = useStudioEnvironmentParams();
   const { data: optionsLists, status } = useOptionListsQuery(org, app);
 
+  const handleDelete = () => {
+    if (component.options) {
+      delete component.options;
+    }
+
+    const emptyOptionsId = '';
+    handleComponentChange({
+      ...component,
+      optionsId: emptyOptionsId,
+    });
+
+    setComponentHasOptionList(false);
+  };
+
```

Comment:
Is a better name for this function: `handleDeleteManualOptions`? Since it only sets the optionsListId? But why is it passed to both editors then? Something is a bit off here for me 😮

Reply:
Ah, I see now. Both properties are set to empty states. But why is `optionsId` property not deleted but set to an empty string?

Reply:
I'll try to remove it and see how it works

Reply:
Nice! 👏 
A general approach is that we never add properties to the layout components that is not used by the app 🤝